xcape/core.py

```diff
@@ -129,40 +129,78 @@ def calc_cape(p, t, td, ps, ts, tds, **kwargs):
     zMUlev : array-like
         height of MUlev (m)
      """
-    
+    if len(args)>1:
+        print('too many optional arguments')
+        
+    elif kwargs['vertical_lev']!=True:
+        print('please select type of vertical grid, pressure level, or model level')
+        
+    elif (len(args)==0)&(kwargs['vertical_lev'] == 'pressure'):
+        print('vertical grid is set to pressure levels,\n'+
+                +'but location in fortran values (1: nlev) of where p <= ps is missing')
+        
+    elif (len(args)==1)&(args[0].shape!=ps.shape)&(kwargs['vertical_lev'] == 'pressure'):
+        print('vertical grid is set to pressure levels,\n'+
+                +'but location in fortran values (1: nlev) of where p <= ps is not the correct shape\n'+
+                +'(pres_lev_pos.shape should be equal to ps.shape)')
+        
+        
     if _any_dask_array(p, t, td, ps, ts, tds):
-        return _calc_cape_gufunc(p, t, td, ps, ts, tds, **kwargs)
+        return _calc_cape_gufunc(p, t, td, ps, ts, tds, *args, **kwargs)
     else:
-        return _calc_cape_numpy(p, t, td, ps, ts, tds, **kwargs)
+        return _calc_cape_numpy(p, t, td, ps, ts, tds, *args, **kwargs)
     
 
-def _calc_cape_gufunc(p, t, td, ps, ts, tds, **kwargs):
-    signature = "(i),(i),(i),(),(),()->(),()"
-    output_dtypes = ('f4','f4')
-    if kwargs['source']=='most-unstable':
-        signature += ",(),()"
-        output_dtypes = output_dtypes + ('i4','f4')
-        
-    return da.apply_gufunc(_calc_cape_numpy, signature,
-                           p, t, td, ps, ts, tds,
-                           output_dtypes=('f4','f4'),
-                           axis=-1,
-                           vectorize=False,
-                           **kwargs)
-
+def _calc_cape_gufunc(p, t, td, ps, ts, tds, *args, **kwargs):
+    
+    if (kwargs['vertical_lev']=='sigma'):
+        signature = "(i),(i),(i),(),(),()->(),()"
+        output_dtypes = ('f4','f4')
+        if kwargs['source']=='most-unstable':
+            signature += ",(),()"
+            output_dtypes = output_dtypes + ('i4','f4')
+
+        return da.apply_gufunc(_calc_cape_numpy, signature,
+                               p, t, td, ps, ts, tds, 
+                               output_dtypes=('f4','f4'),
+                               axis=-1,
+                               vectorize=False,
+                               **kwargs)
+    
```

Comment:
so this is the signature and call for the case for which vertical_lev is sigma, which is your code.